src/Sylius/Component/Core/Promotion/Action/DiscountPromotionActionCommand.php

```diff
@@ -55,6 +56,35 @@ protected function isSubjectValid(PromotionSubjectInterface $subject): bool
         return 0 !== $subject->countItems();
     }
 
+    protected function distributeWithMinimumPrice(array $splitPromotion, array $itemTotals, array $minimumPrices): array
```

Comment:
This method still looks to me like bad design. Why can't this logic be part of one of our distributors or why can't we create a new service for it and deprecate current ones? End users still needs to remember about this function usage if they will extend `DiscountPromotionActionCommand` which for me provides really poor DX

Reply:
its been moved to different service